src/node/progress_tracker.h

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/ccf_assert.h"
+#include "ds/ccf_exception.h"
+#include "kv/kv_types.h"
+#include "kv/tx.h"
+#include "nodes.h"
+#include "tls/tls.h"
+#include "tls/verifier.h"
+
+#include <array>
+#include <vector>
+
+namespace ccf
+{
+  class ProgressTracker
+  {
+  public:
+    ProgressTracker(kv::NodeId id_, ccf::Nodes& nodes_) : id(id_), nodes(nodes_)
+    {}
+
+    void add_signature(
+      kv::Consensus::View view,
+      kv::Consensus::SeqNo seqno,
+      kv::NodeId node_id,
+      uint32_t signature_size,
+      std::array<uint8_t, MBEDTLS_ECDSA_MAX_LEN>& sig)
```

Comment:
Should this be a `ccf::Signature` as defined in `signatures.h`?

Reply:
ccf::Signature is not the best named class because it is more than just a signature it contains other information.

Also this being passed the raw format from the network message. If we just pass this in a different format then we would have to add additional copies.